telemetry/record.go

```diff
@@ -95,3 +106,28 @@ func (record *Record) ensureEncoded() {
 		record.Serializer.Logger().Error("record_RawBytes_blank")
 	}
 }
+
+func getPayloadWithVin(record *Record) ([]byte, error) {
```

Comment:
nit: how about something like:-
```sh
func (record *Record) getPayloadWithVin() ([]byte, error) {
	switch record.TxType {
	case "alerts":
		message := &protos.VehicleAlerts{}
		err := proto.Unmarshal(record.Payload(), message)
		message.Vin = record.Vin
		payload, _ := proto.Marshal(message)
		return payload, err
	case "errors":
		message := &protos.VehicleErrors{}
		err := proto.Unmarshal(record.Payload(), message)
		message.Vin = record.Vin
		payload, _ := proto.Marshal(message)
		return payload, err
	case "V":
		message := &protos.Payload{}
		err := proto.Unmarshal(record.Payload(), message)
		message.Vin = record.Vin
		payload, _ := proto.Marshal(message)
		return payload, err
	}

	return record.PayloadBytes, nil
}
```

Reply:
Not seeing the difference, I updated to use switch case earlier this morning. Is there something else I'm missing? @agbpatro